src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -163,32 +151,190 @@ SkiaGeometrySource2D BuildGeometry()
 					spriteShape.FillBrush = null;
 				}
 
-				return geometry;
+				spriteShape.Geometry = compositor.CreatePathGeometry(path);
+				var shapeVisual = compositor.CreateShapeVisual();
+				shapeVisual.Shapes.Add(spriteShape);
+
+				sublayers.Add(shapeVisual);
+				parent.Children.InsertAtBottom(shapeVisual);
+
+				owner.ClippingIsSetByCornerRadius = cornerRadius != CornerRadius.None;
+				if (owner.ClippingIsSetByCornerRadius)
+				{
+					parent.Clip = new CompositionGeometricClip() { Geometry = spriteShape.Geometry };
+				}
 			}
+			else
+			{
+				var shapeVisual = compositor.CreateShapeVisual();
+
+				var backgroundShape = compositor.CreateSpriteShape();
+
+				if (background is GradientBrush gradientBackground)
+				{
+					var fullArea = new Rect(
+						area.X + borderThickness.Left,
+						area.Y + borderThickness.Top,
+						area.Width - borderThickness.Left - borderThickness.Right,
+						area.Height - borderThickness.Top - borderThickness.Bottom);
+
+					var insideArea = new Rect(default, fullArea.Size);
+					// var insertionIndex = 0;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5932478)